sources/packages/backend/libs/services/src/report/report.service.ts

```diff
@@ -54,6 +58,28 @@ export class ReportService extends RecordDataModelService<ReportConfig> {
     return unparse(reportData);
   }
 
+  /**
+   * Stream file as downloadable response.
+   * @param response http response as file.
+   * @param reportName report name.
+   * @param fileContent content of the file.
+   */
+  streamFile(response: Response, reportName: string, fileContent: string) {
```

Comment:
The `streamFile` is responsible for converting a `fileContent` string into an HTTP response with proper headers and all.
The idea would be to keep this method at the API level. Even moving it to service please try to move to a controller service level. If not possible please keep it at the API service level.
If the above does not make sense please provide some justification to move this method to the generic service level.